packages/backend/scripts/diffc.py

```diff
@@ -0,0 +1,241 @@
+            else:  # parent process
+                result = os.read(fd, 1024).decode()
+                if "No changes." not in result:
+                    print(f"Comparing {file1} and {file2}")
+                    print(result)
+                    no_changes = False
+
+    if no_changes:
+        print("No changes in proxies.")
+
+
+def diff_implementations(folder1, folder2, common_directories):
+    base_path = ".."
+    print(BOLD + "\nComparing implementations..." + RESET)
+
+    # Determine the project structure
+    # use the first directory to determine the project structure
+    dummy_directory = next(iter(common_directories))
+    subpath1, logical_name1 = get_project_subpath(
+        base_path, folder1, dummy_directory)
+    subpath2, logical_name2 = get_project_subpath(
+        base_path, folder2, dummy_directory)
+
+    # Iterate over directories
+    for directory in common_directories:
+        no_changes = True
+        directory_title = " {0} ".format(directory)
+        num_dashes = (terminal_width - len(directory_title)) // 2
+        print(f"\n{'-' * num_dashes}{directory_title}{'-' * num_dashes}\n")
+
+        # Construct the directory paths for the current directory
+        dir_path1 = os.path.join(
+            base_path, "discovery", folder1, "ethereum", ".code", directory, subpath1)
+        dir_path2 = os.path.join(
+            base_path, "discovery", folder2, "ethereum", ".code", directory, subpath2)
+
+        if not os.path.exists(dir_path1):
+            print(
+                f"{folder1} does not contain {directory} with project structure {logical_name1}.")
+            continue
+        if not os.path.exists(dir_path2):
+            print(
+                f"{folder2} does not contain {directory} with project structure {logical_name2}.")
+            continue
+
+        sol_files1 = [os.path.join(root, f) for root, dirs, files in os.walk(
+            dir_path1) for f in files if f.endswith('.sol')]
+        sol_files2 = [os.path.join(root, f) for root, dirs, files in os.walk(
+            dir_path2) for f in files if f.endswith('.sol')]
+
+        # Map to common format
+        sol_files1 = {f.replace(dir_path1, "<placeholder>")
+                      for f in sol_files1}
+        sol_files2 = {f.replace(dir_path2, "<placeholder>")
+                      for f in sol_files2}
+
+        common_sol_files = sol_files1 & sol_files2
+
+        for sol_file in common_sol_files:
+            file1 = sol_file.replace(
+                "<placeholder>", dir_path1).replace(folder1, folder1)
+            file2 = sol_file.replace(
+                "<placeholder>", dir_path2).replace(folder1, folder2)
+
+            # Run "forge fmt" on each file before diffing
+            # subprocess.run(["forge", "fmt", file1])
+            # subprocess.run(["forge", "fmt", file2])
+
+            pid, fd = pty.fork()
+            if pid == 0:  # child process
+                os.execvp("difft", ["difft", "--skip-unchanged",
```

Comment:
this assumes that the user has this executable on his computer. There is no check and a friendly error print out when such an executable is not found. Better yet for portability, diffing tools are not that complex, one can implement a simple diff routine in python at most two hounded lines, look at [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) if you're interested

Reply:
i particularly want to use difftastic since it is syntax-aware: for example, it has explicit solidity supports and for this reason it shows far better diffs than other tools that we have used in the past (either online or internal tools)

regarding the check, i will add a friendly error

Reply:
done